app/jobs/hyrax/grant_edit_to_members_job.rb

```diff
@@ -1,15 +1,16 @@
 module Hyrax
   # Grants edit access for the supplied user for the members attached to a work
-  class GrantEditToMembersJob < ApplicationJob
+  class GrantEditToMembersJob <  Hyrax::ApplicationJob
     queue_as Hyrax.config.ingest_queue_name
 
     # @param [ActiveFedora::Base] work - the work object
     # @param [String] user_key - the user to add
-    def perform(work, user_key)
+    # @param [Boolean] use_valkyrie - use valkyrie objects for this operation?
+    def perform(work, user_key, use_valkyrie: Hyrax.config.use_valkyrie?)
```

Comment:
Could this one branch based on what kind of `work` you pass it? 

The caller is going to have to give a `Valkyrie::Resource` *and* specify `use_valkyrie`, as things stand, right?

Reply:
I had a feeling we should be doing it more dynamically. Changed.